### PR TITLE
Integrate lullabot/drainpipe-installer into this repository

### PR DESCRIPTION
**Pull request by justafish**
It seems like we can integrate the composer plugins directly into this repository just fine (unless I'm missing something!)

- Moves the code from https://github.com/Lullabot/drainpipe-installer to here (collapsing the Taskfile installer into one class)
- Updates the gitignore file creation to just check if the text `.task` is in there if `.gitignore` already exists
- Adds a test for the binary installer
- Hooks up the test to GitHub Actions
- Removes the symlink of Taskfile.yml because it's annoying for local testing at the moment, but will follow up on this :smile:

### CODE REVIEW

**Comment by deviantintegral** [comment]
I was under the impression it had to be a separate repository so the plugin was installed first and available before other code used it. Otherwise, how would we specify both the type of `composer-plugin` and `drainpipe-installer` in composer.json?

**Comment by justafish** [comment]
@deviantintegral seems to work fine. I've tested it locally with a fresh Drupal install and hooked the tests up to GitHub actions

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
+; top-most EditorConfig file
```
Comment (deviantintegral) [inline .editorconfig:1]: 👏

File: .github/workflows/composer-plugin.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    types: [opened, synchronize, reopened]
+
+jobs:
+  Test-Composer-Plugin:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.4'
```
Comment (deviantintegral) [inline .github/workflows/composer-plugin.yml:16]: Not a blocker for this PR, but we could use 8?

File: .php-cs-fixer.dist.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
+<?php
```
Comment (deviantintegral) [inline .php-cs-fixer.dist.php:1]: 👏 👏

File: tests/src/Functional/InstallerScaffoldTest.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        parent::setUpBeforeClass();
+        $composer = new Process(['composer', 'update'], self::PROJECT_PATH);
+        $composer->run();
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        parent::tearDownAfterClass();
+
+        //if (file_exists(self::PROJECT_PATH.'/.gitignore')) {
```
Comment (deviantintegral) [inline tests/src/Functional/InstallerScaffoldTest.php:25]: delete this?

**Review by deviantintegral — APPROVED** [review]
Well, this works perfectly! Glad you saw we could collapse into one project.

I saw we are committing composer.lock - that should probably be removed, so tests follow current dependencies?

**Pull request merged**